components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/OASParserUtil.java

```diff
@@ -1211,4 +1205,35 @@ static JsonNode getResponseCacheConfig(String responseCache, int cacheTimeout) {
         scopesSortedlist.sort(Comparator.comparing(Scope::getName));
         return new LinkedHashSet<>(scopesSortedlist);
     }
-}
+
+    /**
+     * Preprocessing of scopes schemes to support multiple schemes other than 'default' type
+     * This method will change the given definition
+     *
+     * @param swaggerContent
+     * @param apiToAdd
+     * @return processedSwaggerContent
+     */
+    public static String preProcess(String swaggerContent, API apiToAdd) throws APIManagementException {
+        //Load required properties from swagger to the API
+        APIDefinition apiDefinition = getOASParser(swaggerContent);
+        SwaggerVersion swaggerVersion = getSwaggerVersion(swaggerContent);
+        if (swaggerVersion == SwaggerVersion.SWAGGER) {
+            Set<Scope> scopes = OAS2Parser.injectOtherScopesToDefault(swaggerContent);
+            Set<URITemplate> urlTemplates = OAS2Parser.injectOtherRescouceScopesToDefault(swaggerContent);
```

Comment:
injectOtherRescouceScopesToDefault can be moved to the APIDefinition interface. Then this can be implemented by both OAS2Parser and OAS3Parser parser as public methods (not static). After that, this can be called as apiDefinition.injectOtherRescouceScopesToDefault(swaggerContent);. We'll not need this `if` block to check for SwaggerVersion.SWAGGER and have two branches.